### PR TITLE
Replace `hidden service` with `onion service`

### PR DESCRIPTION
**Pull request by RiccardoMasutti**
For a couple of years, Tor has made the term `hidden service` obsolete, in favor of `onion service`: [Tor Project | Onion Services](https://community.torproject.org/onion-services/)

This PR updates all the references.

### CODE REVIEW

**Comment by hebasto** [comment]
Could it be done with a [scripted diff](https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#scripted-diffs)?

**Comment by jonatack** [comment]
Beyond the question of whether this is a change worth making and worth the review time it is requesting, it looks like this PR is a drive-by search and replace that touches a number of files that are auto-generated or not usually changed.

**Comment by Saibato** [comment]
thx @RiccardoMasutti , for pointing this out and I somewhat like this kind of compliance on paper since disguise is the the sister of hidden,

,but as long 

tor source code;
./src/feature/rend/rendservice.h:

``` Code
  /* Other fields */
  crypto_pk_t *private_key; /**< Permanent hidden-service key. */
  char service_id[REND_SERVICE_ID_LEN_BASE32+1]; /**< Onion address without
                                                  * '.onion' */
  char pk_digest[DIGEST_LEN]; /**< Hash of permanent hidden-service key. */
  smartlist_t *intro_nodes; /**< List of rend_intro_point_t's we have,
                             * or are trying to establish. */
  /** List of rend_intro_point_t that are expiring. They are removed once
   * the new descriptor is successfully uploaded. A node in this list CAN
   * NOT appear in the intro_nodes list. */

```
names it, what it should be ?

So imho facts become not that different by wording.
But as this is kind of compliance on paper and disguise the the sister of hidden,
i somewhat like it.  

The moment even 
`HiddenServiceDir `parameter for use in torrc get renamed we should act.

File: src/qt/locale/bitcoin_en.ts [hunk 1]
```diff
@@ -1438,7 +1438,7 @@
     </message>
```
Comment (laanwj) [inline src/qt/locale/bitcoin_en.ts:1438]: Please do not make manual changes to this file

File: doc/release-notes/release-notes-0.14.0.md [hunk 2]
```diff
@@ -689,7 +689,7 @@ and git merge commit are mentioned.
 - #8567 `85d4e21` Add default port numbers to REST doc (djpnewton)
```
Comment (laanwj) [inline doc/release-notes/release-notes-0.14.0.md:689]: Please don't touch historical release notes.

**Comment by laanwj** [comment]
Concept ACK. I think 'onion service' describes more concretely than 'hidden service' what is being talked about.

File: doc/man/bitcoin-qt.1 [hunk 3]
```diff
@@ -193,7 +193,7 @@ Accept connections from outside (default: 1 if no \fB\-proxy\fR or \fB\-connect\
 .HP
 \fB\-listenonion\fR
 .IP
-Automatically create Tor hidden service (default: 1)
+Automatically create Tor onion service (default: 1)
```
Comment (jonatack) [inline doc/man/bitcoin-qt.1:196]: The first line of each of the `man` files states: `DO NOT MODIFY THIS FILE!  It was generated by...`

**Comment by practicalswift** [comment]
Concept ACK assuming @laanwj's feedback is addressed

Welcome as a contributor @RiccardoMasutti! :)

**Comment by fanquake** [comment]
You need to [squash your changes](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits) into a single commit, and use a proper commit message. i.e:

```bash
doc: replace hidden service with onion service

For a couple of years, Tor documentation has made 
the term hidden service obsolete, in favor of onion 
service.

This PR updates all the references in the code base.
```

**Comment by RiccardoMasutti** [comment]
@fanquake @laanwj It should be okay now. Can you review it again please?

Ps: it's my first time squashing commits via git, and I'm really happy to have learned it. Usually I relied on GitHub repo maintainers who have the ability to do `merge and squash`

File: doc/tor.md [hunk 4]
```diff
@@ -110,7 +110,7 @@ Connecting to Tor's control socket API requires one of two authentication method
 configured. It also requires the control socket to be enabled, e.g. put `ControlPort 9051`
 in `torrc` config file. For cookie authentication the user running bitcoind must have read
 access to the `CookieAuthFile` specified in Tor configuration. In some cases this is
-preconfigured and the creation of a hidden service is automatic. If permission problems
+preconfigured and the creation of a onion service is automatic. If permission problems
```
Comment (jonatack) [inline doc/tor.md:113]: s/a onion/an onion/ here and also lines 22 and 101

**Comment by practicalswift** [comment]
ACK f2cd7a4dc93ac117438887cabe6dd095ecf611aa -- patch looks correct

**Review by hebasto — CHANGES_REQUESTED** [review]
Please change here: https://github.com/bitcoin/bitcoin/blob/19f812c57b420ae3e9ab5a95335e2adb5f54faf0/src/qt/forms/optionsdialog.ui#L460-L467

**Comment by laanwj** [comment]
> Welcome as a contributor @RiccardoMasutti! :)

Yes and thanks for sticking with this, most admit that it's not really documented anywhere which files you are and are not allowed to change.

**Comment by RiccardoMasutti** [comment]
@laanwj ready to merge

**Review by hebasto — APPROVED** [review]
ACK 1e72b68ab330c72644981508c8a1b3fa670d086f, tested on Linux Mint 20 (x86_64).

The GUI-related changes:
- master: 
DeepinScreenshot_select-area_20200807194930
- this PR:
DeepinScreenshot_select-area_20200807194725

**Comment by laanwj** [comment]
Code review ACK 1e72b68ab330c72644981508c8a1b3fa670d086f

**Pull request merged**